### PR TITLE
The basic features implementation of discovery probe

### PR DESCRIPTION
**Pull request by hu-weihua**
This pull request include below changes:
1. Implement discovery probe basic feature
2. enhance xcatporbe main framework

Note: the function ``dhcp_dynamic_range_check`` has been done by @xuweibj , she will commit this part in another pull request.

Below are some real output of discovery, it's maybe helpful for you to review:

        [root@fs1 probe]# ./xcatprobe  discovery -m
        Start to capture every message during discovery process......
        Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.153 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.153 via enP4p1s0f2
        Send DHCPACK on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Via HTTP 192.168.5.153 download /tftpboot/pxelinux.cfg/p/192.168.0.0_16
        Via HTTP 192.168.5.153 download //tftpboot/xcat/genesis.kernel.ppc64
        Via HTTP 192.168.5.153 download //tftpboot/xcat/genesis.fs.ppc64.gz
        Send DHCPACK on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.151 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.151 via enP4p1s0f2
        Send DHCPACK on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Via HTTP 192.168.5.151 download /tftpboot/pxelinux.cfg/p/192.168.0.0_16
        Via HTTP 192.168.5.151 download //tftpboot/xcat/genesis.kernel.ppc64
        Via HTTP 192.168.5.151 download //tftpboot/xcat/genesis.fs.ppc64.gz
        Send DHCPACK on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.156 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.156 via enP4p1s0f2
        Send DHCPACK on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPACK on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Acquired IPv4 address
        on enP5p7s0f0
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Run getcert 192.168.3.29:3001...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Getcert done.
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Run getdestiny
        192.168.3.29:3001...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Getdestiny done.
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.154 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.154 via enP4p1s0f2
        Send DHCPACK on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPACK on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: The destparameter
        is discover
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: The destiny is
        discover
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: The dest is discover
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Run dodiscovery...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Beginning
        node discovery process...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Waiting for
        nics to get addresses
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Network configuration
        complete, commencing transmit of discovery packets
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Beginning
        echo infomation to discovery packet file...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Discovery
        packet file is ready.
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Sending the
        discovery packet to xCAT (192.168.3.29:3001)...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Sleeping 5
        seconds...
        Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Restart...
        Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.29 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.29 via enP4p1s0f2
        Send DHCPACK on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Send DHCPACK on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Acquired IPv4 address
        on enP5p7s0f0
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Run getcert 192.168.3.29:3001...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Getcert done.
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Run getdestiny
        192.168.3.29:3001...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Getdestiny done.
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: The destparameter
        is discover
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: The destiny is
        discover
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: The dest is discover
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Run dodiscovery...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Beginning
        node discovery process...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Waiting for
        nics to get addresses
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Network configuration
        complete, commencing transmit of discovery packets
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Beginning
        echo infomation to discovery packet file...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Discovery
        packet file is ready.
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sending the
        discovery packet to xCAT (192.168.3.29:3001)...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sleeping 5
        seconds...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Beginning
        echo infomation to discovery packet file...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Discovery
        packet file is ready.
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sending the
        discovery packet to xCAT (192.168.3.29:3001)...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sleeping 5
        seconds...
        Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Restart...
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        Via HTTP 192.168.5.31 download /tftpboot/petitboot/c910f05c31
        Via HTTP 192.168.5.31 download /tftpboot/xcat/osimage/ubuntu16.04-ppc64el-install-compute/vmlinuz
        Via HTTP 192.168.5.31 download /tftpboot/xcat/osimage/ubuntu16.04-ppc64el-install-compute/initrd.img
        Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        ^C=======================================
        = The summary of discovery:
        =======================================
        [70:e2:84:14:0a:29 (c910f05c31)]
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.151 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.151 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Via HTTP 192.168.5.151 download /tftpboot/pxelinux.cfg/p/192.168.0.0_16
        | Via HTTP 192.168.5.151 download //tftpboot/xcat/genesis.kernel.ppc64
        | Via HTTP 192.168.5.151 download //tftpboot/xcat/genesis.fs.ppc64.gz
        | Send DHCPACK on 192.168.5.151 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.154 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.154 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.154 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Acquired IPv4
        address on enP5p7s0f0
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Run getcert 192.168.3.29:3001...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Getcert done.
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Run getdestiny
        192.168.3.29:3001...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Getdestiny done.
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: The destparameter
        is discover
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: The destiny is
        discover
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: The dest is discover
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.doxcat: Run dodiscovery...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Beginning
        node discovery process...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Waiting
        for nics to get addresses
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Network
        configuration complete, commencing transmit of discovery packets
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Beginning
        echo infomation to discovery packet file...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Discovery
        packet file is ready.
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sending
        the discovery packet to xCAT (192.168.3.29:3001)...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sleeping
        5 seconds...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Beginning
        echo infomation to discovery packet file...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Discovery
        packet file is ready.
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sending
        the discovery packet to xCAT (192.168.3.29:3001)...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Sleeping
        5 seconds...
        | Recv from 70:e2:84:14:0a:29(192.168.5.154) : xcat.genesis.dodiscovery: Restart...
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:29 for 192.168.5.31 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        | Via HTTP 192.168.5.31 download /tftpboot/petitboot/c910f05c31
        | Via HTTP 192.168.5.31 download /tftpboot/xcat/osimage/ubuntu16.04-ppc64el-install-compute/vmlinuz
        | Via HTTP 192.168.5.31 download /tftpboot/xcat/osimage/ubuntu16.04-ppc64el-install-compute/initrd.img
        | Send DHCPACK on 192.168.5.31 back to 70:e2:84:14:0a:29 via enP4p1s0f2
        [70:e2:84:14:0a:05 (c910f05c29)]
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.153 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.153 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Via HTTP 192.168.5.153 download /tftpboot/pxelinux.cfg/p/192.168.0.0_16
        | Via HTTP 192.168.5.153 download //tftpboot/xcat/genesis.kernel.ppc64
        | Via HTTP 192.168.5.153 download //tftpboot/xcat/genesis.fs.ppc64.gz
        | Send DHCPACK on 192.168.5.153 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.156 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.156 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.156 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Acquired IPv4
        address on enP5p7s0f0
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Run getcert 192.168.3.29:3001...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Getcert done.
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Run getdestiny
        192.168.3.29:3001...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Getdestiny done.
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: The destparameter
        is discover
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: The destiny is
        discover
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: The dest is discover
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.doxcat: Run dodiscovery...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Beginning
        node discovery process...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Waiting
        for nics to get addresses
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Network
        configuration complete, commencing transmit of discovery packets
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Beginning
        echo infomation to discovery packet file...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Discovery
        packet file is ready.
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Sending
        the discovery packet to xCAT (192.168.3.29:3001)...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Sleeping
        5 seconds...
        | Recv from 70:e2:84:14:0a:05(192.168.5.156) : xcat.genesis.dodiscovery: Restart...
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPDISCOVER from 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPOFFER on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.29 via enP4p1s0f2
        | Receive DHCPREQUEST from 70:e2:84:14:0a:05 for 192.168.5.29 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2
        | Send DHCPACK on 192.168.5.29 back to 70:e2:84:14:0a:05 via enP4p1s0f2

    [root@fs1 probe]# ./xcatprobe  discovery -h
    Usage:
    discovery -h
    discovery -t
    discovery -p [-V] [-T <discovery_type>] [-n <node_range>]
    discovery -m
    discovery [-V] [-T <discovery_type>] [-n <node_range>]

    Description:
    Do probe for discovery process, including pro-check for required configuration and realtime monitor of discovery process.
    If without any option, discovery will do pro-check first, if all checking point pass, then goes to monitor directly.

    Options:
    -h : Get usage information of discovery.
    -t : To verify if discovery can work, reserve option for probe framework.
    -V : Output more information for debug.
    -p : Do pre-check for disvoery, check whether required configuration have been done ahead.
    -T : The type of discovery, the valid values are mtms,switch.
    -n : The range of predefined node, must used with option -m.
    -m : Do realtime monitor for discovery process.

### CODE REVIEW

**Comment by daniceexi** [comment]
Could you confirm why there's conflict for your pull request to merge?

**Comment by daniceexi** [comment]
One general comments is all the 'dhcp' message has double display.

**Comment by hu-weihua** [comment]
To resolve conflicts, close this pull request and re_pull new one.

File: xCAT-probe/lib/perl/probe_utils.pm [hunk 1]
```diff
@@ -338,4 +341,102 @@ sub is_dns_ready {
...
+        ip: The ip address 
+=cut
+
+#------------------------------------------
+sub get_ip_from_hostname{
+    my $hostname = shift;
+    $hostname=shift if(($hostname) && ($hostname =~ /probe_utils/));
+    my $ip = "";
+
+    my @output = `ping -c 1 $hostname 2>&1`;
```
Comment (zet809) [inline xCAT-probe/lib/perl/probe_utils.pm]: I don't think using ping is a good idea, why not using inet_aton and then inet_ntoa?

**Pull request closed without merging**